src/com/mostc/pftt/host/Host.java

```diff
@@ -44,8 +44,9 @@ public boolean isSafePath(String path) {
 			return false;
 		String pftt_dir = getPfttDir();
 		if (path.startsWith(pftt_dir)) {
-			// don't delete anything in PFTT dir unless its in cache/working
-			if (!path.startsWith(pftt_dir+"/cache/working/"))
+			// don't delete anything in PFTT dir unless its in cache/working or job_work
+			if (!path.startsWith(pftt_dir+"/cache/working/") &&
+					!path.startsWith(pftt_dir+"\\job_work\\"))
```

Comment:
Is the first path check needed at all, since we move everything to a single work dir?

With the work dir path check - it should be read from the env var, as it's taken from there in first place. Should also replace the path separators to be same and lower case in both for the comparison, at least on Windows.

Thanks.

Thanks.

Reply:
I updated it with your suggestion. Please check. 

The cache folder is technically considered a "safe path" as well. But as you said, if we move everything into a single directory (job_work dir), then this won't be needed.

Reply:
@lavturo thanks for catching up. The other part with lowering and path separators in comparison are still missing. I'd suggest at least `str0.replace("\\", "/").startsWith(str1.replace("\\", "/"))`. Or are those already converted somewhere? Lowering would be wrong on case sensitive FS, that's why I meant at least Windows. But with the path separators - both forward and backslash are actually supported in both Java and PHP.

For the safe path - yeah, lets keep the cache for now. Depending on how the setup is implemented, you could turn it back on later or we'll remove the cache folder from the source tree and then it'll become a "safe path", too.

Thanks.

Reply:
@weltling Updated it again with missing part about comparisons. Moved around some of the code, but should still function the same.